src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -908,29 +919,35 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 		return Object.keys(result);
 	}
 
-	ignoreExtensionRecommendation(extensionId: string): void {
-		/* __GDPR__
-			"extensionsRecommendations:ignoreRecommendation" : {
-				"recommendationReason": { "classification": "SystemMetaData", "purpose": "FeatureInsight", "isMeasurement": true },
-				"extensionId": { "classification": "PublicNonPersonalData", "purpose": "FeatureInsight" }
+	toggleExtensionRecommendation(extensionId: string, shouldRecommend: boolean): void {
+		if (shouldRecommend) {
+			this._globallyIgnoredRecommendations = distinct(<string[]>JSON.parse(this.storageService.get('extensionsAssistant/ignored_recommendations', StorageScope.GLOBAL, '[]')))
+				.map(id => id.toLowerCase()).filter(id => id !== extensionId);
+			this._sessionRestoredRecommendations[extensionId.toLowerCase()] = this._sessionIgnoredRecommendations[extensionId.toLowerCase()];
```

Comment:
When adding to `_sessionRestoredRecommendations`, check first if it exists in `_sessionIgnoredRecommendations`. If it doesn't (this won't happen at the moment, but will happen if anyone calls this method on an extension that wasn't ignored), then we remove it from `_globallyIgnoredRecommendations` anyway to support a future feature if an extension needs to be removed from global ignore list.